pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1456,7 +1457,8 @@ func TestPreFilterState(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			ctx, cancel := context.WithCancel(context.Background())
+			_, ctx := ktesting.NewTestContext(t)
```

Comment:
cc @pohly what's the best practice here, initializing the test context outside the loop or inside as refactored. I guess no big difference but outside sounds better?

Reply:
RE: for isolation, different context for different testcase, then this looks better.

Reply:
Inside. Then each test case gets its own context and per-test output.